### ISSUE TITLE
Deleting All Keys from the events_map Causes Agent to No Longer Detect Actions

### ISSUE THREAD

**Issue by agadient** [aquasecurity/tracee#3699]
## Description

Deleting all the keys from the events_map eBPF map causes the agent to no longer detect future actions. An example program that triggers this issue can be found here: https://github.com/Vali-Cyber/ebpf-attacks/tree/main/delete_keys/

## Output of `tracee version`:

Tracee version: "v0.19.0"

## Output of `uname -a`:

Linux hamden 5.15.0-87-generic #97-Ubuntu SMP Mon Oct 2 21:09:21 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

## Contents of /etc/os-release

hamden@hamden:~$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.1 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"

**Comment by rafaeldtinoco** [comment]
We are aware of that. This is related to https://github.com/aquasecurity/tracee/issues/617. 

Since then we added events such as `security_bpf, security_bpf_map, bpf_attach, security_bpf_prog`. We can only know if any other process has tainted our execution and provide appropriate feedback (as an event).

@AlonZivony this could be a good signature for us to open-source, in response to this issue. Not sure that "avoiding" a full privileged user to clear up our maps is something we should try for now (?).

**Comment by rafaeldtinoco** [comment]
@agadient so, are you differentiating the attacks from privileged and non-privileged point of view ? Because any runtime tool, from the moment user has privileges, is already condemned (but alerting end-user about this is something we can/do).

Now, stressing resources as a regular user to avoid a detection to happen, this is a different (and more important) thing I believe, do you agree ?

Just curious. Thanks.

**Comment by AlonZivony** [comment]
> @AlonZivony this could be a good signature for us to open-source, in response to this issue. Not sure that "avoiding" a full privileged user to clear up our maps is something we should try for now (?).

I don't think that this is a wise idea. If someone is trying to deliberately bypass Tracee, than he will probably go over all of the OS signatures as well. This will make it more probable to bypass them, as in the end, in the kernel we are on even capabilities with other kernel modules and eBPF programs.
@agadient FYI, you could as easily just turn off BPF on the machine. There are multiple ways to do it, and it will be as effective. But like @rafaeldtinoco has written, in the game of behavior monitoring, you will probably fall to at least one malicious behavior on your way to sabotage Trace. So you might win the machine, but in a cloud environment in which its very easy to restart a node, this is losing the war.
I hope that in the near future Tracee will also have enforcement capabilities, which will make it possible to prevent such attacks and not only detect them.

**Comment by AlonZivony** [comment]
@agadient thanks for the idea and the code you donated.
We might be aware of the fact that it is possible, but having and OS code that do it reduce the time to create e2e tests for it and maybe in the future we will have some OS solutions for these kind of attacks. We just need to figure out how to publish such things wisely, without helping attackers to further bypass eBPF tools.

**Comment by rafaeldtinoco** [comment]
> I don't think that this is a wise idea. If someone is trying to deliberately bypass Tracee, than he will probably go over all of the OS signatures as well. This will make it more probable to bypass them, as in the end, in the kernel we are on even capabilities with other kernel modules and eBPF programs.

Check https://github.com/aquasecurity/tracee/issues/3698#issuecomment-1817865749. We can discuss that particular case there, but the idea I mentioned was: ok, lots of files being created could be a signature, lots of X being done to taint tracee could be another signature. I think its valid to consider detecting behaviors that could cause harm do tracee (at least to raise a flag to end user).

**Comment by yanivagman** [comment]
Dup of #617
